Fix module dependencies

neithernut opened the pull request:
This PR moves a few things around, breaking circular dependencies, so we have a nice dependency tree.

This is done by:
- [X] Make "command" _not_ depend on "connection"
- [X] Make "command" _not_ depend on "action"
- [X] Make action manager's context settable from the outside on init

If you see any circular dependencies which I didn't yet mention, notify me.

Commit:
Move command processor away from the commands

Commit:
Move config file opener to connection subsystem

Commit:
Remove old dependencies from test CMakeLists

Commit:
Add context to commands intended for the command list

Commit:
Add function for adding commands to the list

Commit:
Lookup commands in new list

Commit:
Add old list to new one, dynamically

Commit:
Move commands controling a processor to "action"

They are now added dynamically, which solves a dependency issue.

Commit:
Initialize command subsystem from main()

matthiasbeyer commented on src/command/command.c:
```diff
@@ -86,3 +132,53 @@ ws_command_get(
 }
 
+int
+ws_command_add(
```
This should be a plural (`ws_commands_add` instead of `ws_command_add`)

neithernut replied:
The file's prefix is `ws_command`. If anything, I the entire file should be renamed and **all** the prefixes changed.
I _can_ do this, but I'd rather not.

matthiasbeyer replied:
Oh shit. I wouldn't rename the functions either, but maybe we could use something like `ws_command_add_commands` here? Would be a bit more verbose about what happens...

neithernut replied:
Nah... I mean: what's going to be added apart from... commands?

matthiasbeyer replied:
Good point.

Commit:
Fix type of "name" member in ws_command

neithernut commented:
This is ready for merge now.
Note that the contents of `context.c` will have to be moved at some point too, but after merging this we will _at least_ have a starting point for integrating all the functions we currently have sitting around in PRs.

---

Ah fuck it, I'll move `context.c` now.

Commit:
Make action manager's context object settable on init

Commit:
Moce context to global scope

The context object is an omnipotent object which has influence on quite
a few subsystems and hence pulls in a lot of dependencies. Putting it in
the global scope is the only sane option here.

Commit:
Initialize the action manager

We previously forgot this.

neithernut commented:
_Now_ I'm ready. Everyone working on the context should be rebased once this is merged.

---

A future PR will make all the modules real static libraries btw.

matthiasbeyer commented on src/action/commands.h:
```diff
@@ -0,0 +1,39 @@
+#include "command/command.h"
+
+/**
+ * Add comamnds which have impact to the processor
```
Typo

matthiasbeyer commented on src/action/commands.h:
```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Add comamnds which have impact to the processor
+ */
```
Missing `@return` documentation

TheNeikos commented:
ACK

matthiasbeyer commented:
ACK.
